### PR TITLE
Mysql

### PR DESCRIPTION
**Pull request by turnopil**
Added parameterized class mysql. Now you can set mysql_root_pass, which version to install(5.6 or 5.7) and bind_address + serverid (for master and replication nodes). Rewritten rottpass manifest and added mysql Yumrepo. Cleaned not necessary vars in profiles and made declaration of parameterized mysql class. Some minor fixes

### CODE REVIEW

File: modules/mysql/manifests/configure.pp [hunk 1]
```diff
@@ -6,16 +6,16 @@
 #
 # @example
 #   include mysql::configure
-class mysql::configure (
-  $bind = '0.0.0.0',
-)
+class mysql::configure
 {
   $conf_name = $facts['os']['family'] ? {
```
Comment (ipleten) [inline modules/mysql/manifests/configure.pp]: these things are useless as we use only redhat family.
Remove and hardcode config file.
/etc/my.cnf is default files always.

File: modules/mysql/manifests/db.pp [hunk 2]
```diff
@@ -13,14 +13,14 @@
 )
```
Comment (ipleten) [inline modules/mysql/manifests/db.pp:13]: I would default charset and collate to utf8 and ut8_general_ci

File: modules/mysql/manifests/init.pp [hunk 3]
```diff
@@ -6,7 +6,23 @@
 #
 # @example
 #   include mysql
-class mysql {
+class mysql (
+	$mysql_root_password = undef,
+	$mysql_distro        = "community",
+	$mysql_version       = "5.7",
+	$mysql_serverid      = undef,
+  $bind_address        = "0.0.0.0"
```
Comment (ipleten) [inline modules/mysql/manifests/init.pp]: indentation is not good here.

File: modules/mysql/manifests/init.pp [hunk 4]
```diff
@@ -6,7 +6,23 @@
...
+#	$ensure              = "running",
```
Comment (ipleten) [inline modules/mysql/manifests/init.pp]: Comments should be only comments.
Any commented code should be removed.

File: modules/profile/manifests/mysqlserver/replication.pp [hunk 5]
```diff
@@ -1,25 +1,30 @@
...
   }
 
+class { 'mysql':
+  mysql_root_password => $mysql_root_password,
+  mysql_distro        => $mysql_distro,
+  mysql_version       => $mysql_version,
+  mysql_serverid      => $mysql_serverid,
+  bind_address        => $bind_address,
+}
   if $is_slave {
```
Comment (ipleten) [inline modules/profile/manifests/mysqlserver/replication.pp]: that solution looks not pretty good.
But I can't suggest another one for now...

File: modules/mysql/manifests/install.pp [hunk 6]
```diff
@@ -6,22 +6,29 @@
...
+ 	$basearch   = $hardwaremodel
+
+ 	if $mysql::mysql_distro == "community" {
+    $repo_descr = "MySQL $mysql::mysql_version Community Server"
+    $repo_url   = "http://repo.mysql.com/yum/mysql-${mysql::mysql_version}-community/el/$releasever/$basearch/"
+ 	}
+}
 }
+
+yumrepo { "mysql-repo":
```
Comment (ipleten) [inline modules/mysql/manifests/install.pp]: yum repo definition should be moved to profile level.

**Review by ipleten — CHANGES_REQUESTED** [review]

File: modules/mysql/manifests/init.pp [hunk 7]
```diff
@@ -6,7 +6,23 @@
...
+)
+
+{
+  if $mysql::mysql_version == "5.7" {
+    $mysql_ver="57"
```
Comment (savroman) [inline modules/mysql/manifests/init.pp]: Don't understand where "$mysql_ver" is used later?

Comment (turnopil) [reply modules/mysql/manifests/init.pp]: in Yumrepo, install.pp manifest. I would write some comments for better understanding

File: modules/mysql/manifests/configure.pp (hunk 1 above)
Comment (turnopil) [reply modules/mysql/manifests/configure.pp]: i try to use this module in aws ec2 ubuntu instance. But ok, i will delete this options for now

File: modules/mysql/manifests/db.pp (hunk 2 above)
Comment (turnopil) [reply modules/mysql/manifests/db.pp:13]: Good suggestion, thank you!

File: modules/mysql/manifests/init.pp [hunk 8]
```diff
@@ -7,15 +7,15 @@
...
-	$mysql_root_password = undef,
-	$mysql_distro        = "community",
-	$mysql_version       = "5.7",
-	$mysql_serverid      = undef,
-  $bind_address        = "0.0.0.0"
+$mysql_root_password = undef,
+$mysql_distro        = "community",
+$mysql_version       = "5.7",
+$mysql_serverid      = undef,
+$bind_address        = "0.0.0.0"
```
Comment (ipleten) [inline modules/mysql/manifests/init.pp]: 2 spaces is a good indentation for each inner  level.

File: modules/profile/manifests/mysqlserver/replication.pp [hunk 9]
```diff
@@ -26,8 +26,13 @@
   bind_address        => $bind_address,
 }
 
+$repo_descr = $mysql::install::repo_descr
```
Comment (ipleten) [inline modules/profile/manifests/mysqlserver/replication.pp]: Variables shouldn't go from module.
All data should be defined on PROFILE level.

**Review by ipleten — CHANGES_REQUESTED** [review]

File: modules/profile/manifests/mysqlserver/replication.pp [hunk 10]
```diff
@@ -11,28 +11,31 @@
...
+  dports => $port,
+}
 
-class { 'mysql':
-  mysql_root_password => $mysql_root_password,
-  mysql_distro        => $mysql_distro,
-  mysql_version       => $mysql_version,
-  mysql_serverid      => $mysql_serverid,
-  bind_address        => $bind_address,
+package { 'yum':
```
Comment (ipleten) [inline modules/profile/manifests/mysqlserver/replication.pp]: I didn't  get why we need that?
Is yum absent on centos 7 by default?

**Review by ipleten — APPROVED** [review]
Changes looks good to me except that "yum" thing. Explain why?

File: modules/profile/manifests/mysqlserver/replication.pp (hunk 10 above)
Comment (turnopil) [reply modules/profile/manifests/mysqlserver/replication.pp]: It's a trick how to run yumrepo first in all of manifests. Without it want work property, but i will try to find another solution.

**Comment by turnopil** [comment]
It's a trick how to run yumrepo first in all of manifests. Without it want work property, but i will try to find another solution.

**Review by savroman — APPROVED** [review]
I think we can merge it

**Pull request merged**